### PR TITLE
Fix handling of `java.lang.Void` and enable a couple more Error Prone checks

### PR DESCRIPTION
**Pull request by msridhar**
We now require the developer to write `@Nullable` on usages of `java.lang.Void`, consistent with JSpecify (see https://github.com/jspecify/jspecify/issues/51).

This issue was discovered while enabling two new Error Prone checks,
[TypeToString](http://errorprone.info/bugpattern/TypeToString) and [SymbolToString](http://errorprone.info/bugpattern/SymbolToString).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -72,6 +72,8 @@ subprojects { project ->
             check("StringSplitter", CheckSeverity.OFF)
             check("WildcardImport", CheckSeverity.ERROR)
             check("MissingBraces", CheckSeverity.ERROR)
+            check("TypeToString", CheckSeverity.ERROR)
+            check("SymbolToString", CheckSeverity.ERROR)
```
Comment (lazaroclapp) [inline build.gradle:76]: Interesting, these are all suggestion-level, so even with `-Werror`, we don't get them by default...

**Review by lazaroclapp — APPROVED** [review]

File: nullaway/src/main/java/com/uber/nullaway/NullAway.java [hunk 2]
```diff
@@ -178,6 +179,8 @@
   // Unmatched, used for when we only want full checker suppressions to work
   static final String CORE_CHECK_NAME = "NullAway.<core>";
 
+  private static final Supplier<Type> JAVA_LANG_VOID = Suppliers.typeFromString("java.lang.Void");
```
Comment (Stephan202) [inline nullaway/src/main/java/com/uber/nullaway/NullAway.java:182]: There's also `Suppliers.VOID_TYPE`, which looking at the implementation may be more efficient. (Uses `state.getSymtab().voidType`.)

Comment (Stephan202) [reply nullaway/src/main/java/com/uber/nullaway/NullAway.java:182]: Oh, or is that the non-boxed variant? In that case, nevermind.

Comment (msridhar) [reply nullaway/src/main/java/com/uber/nullaway/NullAway.java:182]: There is `Suppliers#JAVA_LANG_VOID_TYPE` which I think does what we want.  Thanks for the tip!

File: build.gradle (hunk 1 above)
Comment (msridhar) [reply build.gradle:76]: Yup.  I found these just browsing through the available checks one day a while back

File: nullaway/src/main/java/com/uber/nullaway/NullAway.java (hunk 2 above)
Comment (Stephan202) [reply nullaway/src/main/java/com/uber/nullaway/NullAway.java:182]: Missed that one :facepalm:. Nice!

Comment (msridhar) [reply nullaway/src/main/java/com/uber/nullaway/NullAway.java]: In the end I realized we really meant to be checking for the unboxed type!  🙂 And there is an `isPrimitiveOrVoid()` method for doing that.

**Comment by msridhar** [comment]
We now report an error for this code in `NullAwayNegativeCases`:

https://github.com/uber/NullAway/blob/4a426db34db58cebb179053fd110b110e7aaa29d/nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayNegativeCases.java#L440-L442

I think this is just wrong and we messed up before.  In fact, since the return type is `java.lang.Void`, this method can _only_ return `null` (since there are no other values assignable to `Void`), so the return _should_ be annotated `@Nullable`.  Do you agree @lazaroclapp?

**Comment by lazaroclapp** [comment]
> I think this is just wrong and we messed up before. In fact, since the return type is `java.lang.Void`, this method can _only_ return `null` (since there are no other values assignable to `Void`), so the return _should_ be annotated `@Nullable`. Do you agree @lazaroclapp?

Arguably `Void == @Nullable Void` as `@NonNull Void` is an empty/error/bottom type... but I wonder how much would need to change in NullAway for there to be a "default-nullable" type/class, vs just saying that if you are returning `Void`, you must explicitly say `@Nullable Void`...

**Comment by lazaroclapp** [comment]
@msridhar JSpecify currently says "No special treatment" for `Void`. So that's good enough for me. `@Nullable Void` it is.

Is this technically the first NullAway question, unrelated to direct efforts to support JSpecify, that is settled by referring to the standard? 😉 

See. https://github.com/jspecify/jspecify/issues/51

**Comment by msridhar** [comment]
> Is this technically the first NullAway question, unrelated to direct efforts to support JSpecify, that is settled by referring to the standard? 😉

First time!  So cool to have a standard 😄

**Comment by msridhar** [comment]
Asking for a new review @lazaroclapp since the main purpose of this PR has changed 🙂 We should note this change in the release notes, as after updating to the next release, users will have to add `@Nullable` to any `java.lang.Void` return type

**Review by lazaroclapp — APPROVED** [review]
LGTM

**Pull request merged**